BOJ/BOJ_1167.py

```diff
@@ -0,0 +1,33 @@
+import sys
+from collections import deque
+input = sys.stdin.readline
+
+
+def bfs(start):
+    visited = [-1 for _ in range(V+1)]
+    queue = deque([start])
+    visited[start] = 0
+    maxi = (0, 0)  # 거리, 노드번호
```

Comment:
```suggestion
    maxi = [0, 0]
```
아래서 보니까 maxi에 계속 튜플을 대입하던데 그것보단 mutable 변수인 list나 변수를 따로 관리하는게 좋아요.
왜냐면 매번 아래서 튜플로 갈아끼우게 되면 이전에 메모리에 있던 튜플값들은 가비지가 되어서 쌓이게 됩니다!!
(코딩테스트에서 메모리 초과를 발생시키는 원인이 될 수 있어요..)
변경 O -> mutable (list, dict, set)
변경 X -> immutable (tuple, str)

Reply:
헉 대박 전혀 몰랐던...
그럼 tuple은 그냥 사용을 안한다고 생각하는게 좋나요?

Reply:
아녀 내부가 변할일이 없으면 튜플로 많이 이용해요
예를 들면 heapq에 데이터를 넣고 뺄때 내부 데이터가 변하지 않는다면 튜플을 이용하는게 효율적이겠죠?